src/test/smart-components/addRoleWizard/inventory-groups-role.test.js

```diff
@@ -0,0 +1,157 @@
+  inventoryReducer: {
+    isLoading: false,
+    resourceTypes: {
+      'inventory:hosts:read': testInventoryGroups.reduce((acc, curr) => ({ ...acc, [curr.name]: { ...curr } }), {}),
+      'inventory:hosts:write': testInventoryGroups.reduce((acc, curr) => ({ ...acc, [curr.name]: { ...curr } }), {}),
+    },
+    total: testInventoryGroups.length,
+  },
+};
+
+jest.mock('@data-driven-forms/react-form-renderer/use-form-api', () => {
+  return {
+    __esModule: true,
+    default: () => ({
+      getState: jest.fn().mockReturnValue({
+        values: {
+          'add-permissions-table': [
+            {
+              uuid: 'inventory:hosts:read',
+              requires: [],
+            },
+            {
+              uuid: 'inventory:groups:write',
+              requires: [],
+            },
+          ],
+          'role-description': 'Be able to read and edit Account Staleness and Culling data.',
+          'role-name': 'Account Staleness and Culling Administrator',
+          'role-type': 'create',
+          'role-uuid': '218ed86a-08f9-4d5d-aeac-7539eed882cd',
+        },
+      }),
+      change: jest.fn(),
+    }),
+  };
+});
+
+jest.mock('@data-driven-forms/react-form-renderer/use-field-api', () => {
+  return {
+    __esModule: true,
+    default: () => ({
+      input: {
+        value: ['inventory:hosts:read', 'inventory:groups:write'],
+        name: 'inventory-groups-role',
+        onChange: jest.fn(),
+        onBlur: jest.fn(),
+        onFocus: jest.fn(),
+      },
+    }),
+  };
+});
+
+const renderComponent = (store) => {
+  return render(
+    <React.Fragment>
+      <Provider store={store}>
+        <BrowserRouter>
+          <InventoryGroupsRole />
+        </BrowserRouter>
+      </Provider>
+    </React.Fragment>
+  );
+};
+
+describe('Inventory groups role', () => {
+  let mockStore;
+
+  const fetchInventoryGroupsSpy = jest.spyOn(InventoryActions, 'fetchInventoryGroups');
+
+  beforeEach(() => {
+    mockStore = configureStore();
+  });
+
+  afterEach(() => {
+    fetchInventoryGroupsSpy.mockReset();
+  });
+
+  test('Add permissions to group renders without failing', async () => {
+    fetchInventoryGroupsSpy.mockImplementationOnce(() => ({
+      type: FETCH_INVENTORY_GROUP,
+      payload: Promise.resolve({}),
+    }));
+    const store = mockStore(initialStateWithPermissions);
+
+    const renderedResults = renderComponent(store);
+
+    expect(renderedResults.getByText('Permissions')).toBeInTheDocument();
+    expect(renderedResults.asFragment()).toMatchSnapshot();
+  });
+
+  test('Display available permission groups successfully', async () => {
+    fetchInventoryGroupsSpy.mockImplementationOnce(() => ({
+      type: FETCH_INVENTORY_GROUP,
+      payload: Promise.resolve({}),
+    }));
+
+    const store = mockStore(initialStateWithPermissions);
+    renderComponent(store);
+
+    fireEvent.click(screen.getAllByLabelText('Options menu')[0]);
+
+    expect(screen.getByText('fooBar')).toBeInTheDocument();
```

Comment:
here we could also check at least last group visible or number of items (in a follow-up also "See more" visibility could be checked as a different test case)

Reply:
another thing to check should be the presence and functionality of the "Select all" option

Reply:
also "Ungrouped systems" option needs to be checked for the host permissions. We'll add it to the follow-up ticket